src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/tabs/AnalyticsTabPane.java

```diff
@@ -1,21 +1,105 @@
+import javafx.scene.chart.Axis;
+import javafx.scene.chart.BarChart;
+import javafx.scene.chart.CategoryAxis;
+import javafx.scene.chart.NumberAxis;
+import javafx.scene.chart.XYChart;
+import javafx.scene.control.Label;
+import javafx.scene.layout.Pane;
 
 /**
  * Tab pane that displays analytics.
  */
 public class AnalyticsTabPane extends UiPart<Canvas> {
+
     private static final String FXML = "AnalyticsTabPane.fxml";
 
+    @FXML
+    private Pane expenseAnalyticsPane;
+
+    @FXML
+    private Pane incomeAnalyticsPane;
+
+    @FXML
+    private Pane savingsAnalyticsPane;
+
+    @FXML
+    private Label expenseAnalyticsLabel;
+
+    @FXML
+    private Label incomeAnalyticsLabel;
+
+    @FXML
+    private Label savingsAnalyticsLabel;
+
+    @FXML
+    private BarChart<String, Number> expenseAnalyticsBarChart;
+
+    @FXML
+    private BarChart<String, Number> incomeAnalyticsBarChart;
+
+    @FXML
+    private BarChart<String, Number> savingsAnalyticsBarChart;
+
     /**
      * Creates an {@code AnalyticsTabPane}.
      */
     public AnalyticsTabPane(MonthlyBudget monthlyBudget) {
         super(FXML);
+
+        Axis<String> expenseAnalyticsStringAxis = new CategoryAxis();
+        Axis<Number> expenseAnalyticsNumberAxis = new NumberAxis();
+        expenseAnalyticsBarChart = new BarChart<>(expenseAnalyticsStringAxis, expenseAnalyticsNumberAxis);
+        expenseAnalyticsBarChart.setLegendVisible(false);
+        expenseAnalyticsBarChart.setVerticalGridLinesVisible(false);
+        expenseAnalyticsBarChart.lookup(".chart-plot-background").setStyle("-fx-background-color: transparent;");
+        expenseAnalyticsPane.getChildren().add(expenseAnalyticsBarChart);
+        expenseAnalyticsLabel.setText("EXPENSE");
+
+        Axis<String> incomeAnalyticsStringAxis = new CategoryAxis();
+        Axis<Number> incomeAnalyticsNumberAxis = new NumberAxis();
+        incomeAnalyticsBarChart = new BarChart<>(incomeAnalyticsStringAxis, incomeAnalyticsNumberAxis);
+        incomeAnalyticsBarChart.setLegendVisible(false);
+        incomeAnalyticsBarChart.setVerticalGridLinesVisible(false);
+        incomeAnalyticsBarChart.lookup(".chart-plot-background").setStyle("-fx-background-color: transparent;");
+        incomeAnalyticsPane.getChildren().add(incomeAnalyticsBarChart);
+        incomeAnalyticsLabel.setText("INCOME");
+
+        Axis<String> savingsAnalyticsStringAxis = new CategoryAxis();
+        Axis<Number> savingsAnalyticsNumberAxis = new NumberAxis();
+        savingsAnalyticsBarChart = new BarChart<>(savingsAnalyticsStringAxis, savingsAnalyticsNumberAxis);
+        savingsAnalyticsBarChart.setLegendVisible(false);
+        savingsAnalyticsBarChart.setVerticalGridLinesVisible(false);
+        savingsAnalyticsBarChart.lookup(".chart-plot-background").setStyle("-fx-background-color: transparent;");
+        savingsAnalyticsPane.getChildren().add(savingsAnalyticsBarChart);
+        savingsAnalyticsLabel.setText("SAVINGS");
```

Comment:
Since these are mostly the same, can they be generalised into a method? Or will these diverse significantly in the future?